04_numeronym/brnbke/main.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"unicode/utf8"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	var result = make([]string, 0, len(vals))
+	for _, val := range vals {
+		l := utf8.RuneCountInString(val)
+		if l > 3 {
+			s := []rune(val)[0]   // starting character
+			e := []rune(val)[l-1] // ending character
+			result = append(result, fmt.Sprintf("%c%d%c", s, l-2, e))
```

Comment:
suggestion: do the `append` once outside the `else` below (get rid of the else) by changing this to:
```
val = fmt.Sprintf("%c%d%c", s, l-2, e)
```
since conceptually what you're doing in this branch is changing the value. An `append` is done for all values.

Reply:
Fair point, actually makes it easier to read also, implemented.